cdap-app-fabric-tests/src/test/java/co/cask/cdap/WorkflowAppWithScopedParameters.java

```diff
@@ -206,21 +206,11 @@ public void destroy() {
     }
   }
 
-  public static class OneAction extends AbstractWorkflowAction {
+  public static class OneAction extends AbstractCustomAction {
 
     @Override
     public void run() {
-      Map<String, WorkflowNodeState> nodeStates = getContext().getNodeStates();
-      Preconditions.checkArgument(2 == nodeStates.size());
-      WorkflowNodeState nodeState = nodeStates.get(ONE_MR);
-      Preconditions.checkArgument(ONE_MR.equals(nodeState.getNodeId()));
-      Preconditions.checkArgument(nodeState.getRunId() != null);
-      Preconditions.checkArgument(NodeStatus.COMPLETED == nodeState.getNodeStatus());
-
-      nodeState = nodeStates.get(ONE_SPARK);
-      Preconditions.checkArgument(ONE_SPARK.equals(nodeState.getNodeId()));
-      Preconditions.checkArgument(nodeState.getRunId() != null);
-      Preconditions.checkArgument(NodeStatus.COMPLETED == nodeState.getNodeStatus());
+      // no-op
```

Comment:
Why this becomes no-op? I thought it is part of the test?

Reply:
`WorkflowContext` was available from `AbstractWorkflowAction` which had the node states. however now `CustomActionContext` will only have the state of the custom action available via `getState`. 

As for the test, workflow node states are available to `destroy` method and we perform checks there already.

Reply:
I see